src/Frontend/Core/Engine/TemplateModifiers.php

```diff
@@ -406,6 +407,8 @@ public static function parseWidget($module, $action, $id = null)
             FrontendModel::getContainer()->set('parseWidget', null);
 
             return $content;
+        } catch (RedirectException $e) {
+            return $e->getResponse();
```

Comment:
Yes, it works. But not perfectly (since there is probably not better fix?).
Check this page above to find out what it does...